barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.hpp

```diff
@@ -328,18 +328,25 @@ acir_format circuit_buf_to_acir_format(std::vector<uint8_t> const& buf)
     return af;
 }
 
+/**
+ * @brief Converts from the ACIR-native `WitnessMap` format to Barretenberg's internal `WitnessVector` format.
+ *
+ * @param buf Serialized representation of a `WitnessMap`.
+ * @return A `WitnessVector` equivalent to the passed `WitnessMap`.
+ * @note This transformation results in all unassigned witnesses within the `WitnessMap` being assigned the value 0.
+ *       Converting the `WitnessVector` back to a `WitnessMap` is unlikely to return the exact same `WitnessMap`.
+ */
 WitnessVector witness_buf_to_witness_data(std::vector<uint8_t> const& buf)
 {
     auto w = WitnessMap::WitnessMap::bincodeDeserialize(buf);
     WitnessVector wv;
     // TODO(https://github.com/AztecProtocol/barretenberg/issues/816): Does "index" need to be initialized to 0 once we
```

Comment:
Can you guys clarify this comment to say whether or not this `index` needs to be decremented by 1 to account for resolution of #3887? I think it does.

Reply:
There's some discussion of this [here](https://github.com/AztecProtocol/aztec-packages/pull/3940#discussion_r1448692631)